public/app/features/inspector/InspectJSONTab.tsx

```diff
@@ -121,6 +122,14 @@ export class InspectJSONTab extends PureComponent<Props, State> {
         } else {
           const updates = JSON.parse(this.state.text);
           dashboard!.shouldUpdateDashboardPanelFromJSON(updates, panel!);
+
+          //Report relevant updates
+          reportInteraction('grafana_panel_inspect_apply_clicked', {
+            panel_type_changed: panel!.type !== updates.type,
+            panel_id_changed: panel!.id !== updates.id,
+            panel_grid_pos_changed: !isEqual(panel!.gridPos, updates.gridPos),
+          });
```

Comment:
@axelavargas about the suggestion of adding metadata about the updating DS configuration, WDYT?
```suggestion
          reportInteraction('grafana_panel_inspect_apply_clicked', {
            panel_type_changed: panel!.type !== updates.type,
            panel_id_changed: panel!.id !== updates.id,
            panel_grid_pos_changed: !isEqual(panel!.gridPos, updates.gridPos),
            panel_datasource_changed: !isEqual(panel!.targets, updates.targets),
          });
```

I have the hypothesis based on my usage, that the JSON inspect changes can be used to "find & replace" queries in batch. Since we agreed we want to monitor in which cases the user edits the JSON, this scenario is valid.